ext/standard/math.c

```diff
@@ -301,19 +304,46 @@ PHP_FUNCTION(floor)
 }
 /* }}} */
 
+int php_math_round_mode_from_enum(zend_object *mode)
+{
+	zval *case_name = zend_enum_fetch_case_name(mode);
+	zend_string *mode_name = Z_STR_P(case_name);
+
+	switch (ZSTR_VAL(mode_name)[0] + ZSTR_VAL(mode_name)[4]) {
+		case 'H' + 'A':
+			return PHP_ROUND_HALF_UP;
+		case 'H' + 'T':
+			return PHP_ROUND_HALF_DOWN;
+		case 'H' + 'E':
+			return PHP_ROUND_HALF_EVEN;
+		case 'H' + 'O':
+			return PHP_ROUND_HALF_ODD;
+		case 'T' + 'r':
+			return PHP_ROUND_TOWARD_ZERO;
+		case 'A' + 'F':
+			return PHP_ROUND_AWAY_FROM_ZERO;
+		case 'N' + 't':
+			return PHP_ROUND_FLOOR;
+		case 'P' + 't':
+			return PHP_ROUND_CEILING;
+		EMPTY_SWITCH_DEFAULT_CASE();
+	}
```

Comment:
Are case names case-sensitive or not? Because that might be a bit of an isssue with the mode name no?

Reply:
This check is equivalent to retrieving the `->name` property of the `RoundingMode` enum, thus it matches the casing defined in the stub. Generally speaking, enum cases are class constants, which are case-sensitive.

Reply:
Right, I forgot that they are class constants